src/backend/dx11/src/device.rs

```diff
@@ -1797,27 +1797,47 @@ impl device::Device<Backend> for Device {
         J::Item: Borrow<pso::Descriptor<'a, Backend>>,
     {
         for write in write_iter {
+            // Get baseline mapping
             let mut mapping = write
                 .set
                 .layout
                 .pool_mapping
                 .map_register(|mapping| mapping.offset);
+
+            // Iterate over layout bindings until the first binding is found.
             let binding_start = write
                 .set
                 .layout
                 .bindings
                 .iter()
                 .position(|binding| binding.binding == write.binding)
                 .unwrap();
+
+            // If we've skipped layout bindings, we need to add them to get the correct binding offset
             for binding in &write.set.layout.bindings[..binding_start] {
                 let content = DescriptorContent::from(binding.ty);
-                mapping.add_content(content, binding.stage_flags);
+                mapping.add_content_many(content, binding.stage_flags, binding.count as _);
             }
 
-            for (binding, descriptor) in write.set.layout.bindings[binding_start..]
-                .iter()
-                .zip(write.descriptors)
-            {
+            // We start at the given binding index and array index
+            let mut binding_index = write.binding as usize;
```

Comment:
this should be using `binding_start`, not `write.binding`. These are different things!